kibble/datastore/file_system_datasource_test.go

```diff
@@ -0,0 +1,99 @@
+//    Copyright 2018 SHIFT72
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+package datastore
+
+import (
+	"testing"
+
+	"kibble/models"
+	"kibble/test"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestRootTemplateType(t *testing.T) {
+
+	var fsDS FileSystemDataSource
+
+	renderer := &test.MockRenderer{}
+
+	r := &models.Route{
+		Name:         "root",
+		URLPath:      "",
+		TemplatePath: "../sample_site/",
+		DataSource:   "FileSystem",
+	}
+
+	ctx := models.RenderContext{
+		Route:       r,
+		RoutePrefix: "",
+		Site:        &models.Site{},
+	}
+	fsDS.Iterator(ctx, renderer)
+
+	assert.True(t, renderer.RenderCalled)
+	assert.Equal(t, "../sample_site/test.html.jet", renderer.TemplatePath)
+	assert.Equal(t, "../sample_site/test.html", renderer.FilePath)
+}
+
+func TestRootTemplateWithRoutePrefixType(t *testing.T) {
+
+	var fsDS FileSystemDataSource
+
+	renderer := &test.MockRenderer{}
+
+	r := &models.Route{
+		Name:         "root",
+		URLPath:      "",
+		TemplatePath: ".",
+		DataSource:   "FileSystem",
+	}
+
+	ctx := models.RenderContext{
+		Route:       r,
+		RoutePrefix: "/fr",
+		Site:        &models.Site{},
+	}
+
+	fsDS.Iterator(ctx, renderer)
+
+	assert.False(t, renderer.RenderCalled)
+}
+
+func TestTemplateWithRoutePrefixType(t *testing.T) {
+
+	var fsDS FileSystemDataSource
+
+	renderer := &test.MockRenderer{}
+
+	r := &models.Route{
+		Name:         "static",
+		URLPath:      "",
+		TemplatePath: "../sample_site/.well-known/",
+		DataSource:   "FileSystem",
+	}
+
+	ctx := models.RenderContext{
+		Route:       r,
+		RoutePrefix: "/fr/blah", // blah added because we have used the relative .. pathing above
+		Site:        &models.Site{},
+	}
+
+	fsDS.Iterator(ctx, renderer)
+
+	assert.True(t, renderer.RenderCalled)
+	assert.Equal(t, "../sample_site/.well-known/example.jet", renderer.TemplatePath)
+	assert.Equal(t, "/fr/sample_site/.well-known/example", renderer.FilePath)
```

Comment:
Would there be a call to not duplicate `.well-known` files across all languages? Wondering if we want to be bit less bloated in what files we create.

Reply:
Yes can be done - naming the route "root", means it will not apply these files to all languages. Bit of an open question if we should apply these to all languages most of the time. `/well-known/` being a good example of routes that we should not replicate. It is a poor example.